src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,150 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> nodeHead;
+    private Node<T> nodeTail;
+    private int size;
+
+    public MyLinkedList() {
+        this.nodeHead = this.nodeTail = new Node(null, null, null);
+        this.size = 0;
+    }
+
+    private class Node<T> {
+        private T element;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(T element, Node<T> prev, Node<T> next) {
+            this.element = element;
+            this.prev = prev;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> currentNode = nodeHead;
+        for (int i = 0; i < size; i++) {
+            currentNode = currentNode.next;
+        }
+        Node<T> newNode = new Node<T>(value, currentNode, null);
+        if (nodeHead == null) {
+            nodeHead = newNode;
+            nodeTail = newNode;
+        } else {
+            newNode.prev = nodeTail;
+            nodeTail.next = newNode;
+            nodeTail = newNode;
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("No such index " + index);
+        }
+        Node<T> currentNode = nodeHead;
+        for (int i = 0; i < index; i++) {
+            currentNode = currentNode.next;
+        }
```

Comment:
lets move logic where we get node by index in separate private method as we use it several times in code